var/spack/repos/builtin/packages/py-scikit-optimize/package.py

```diff
@@ -21,17 +21,24 @@ class PyScikitOptimize(PythonPackage):
     maintainers("liuyangzhuan")
 
     version("master", branch="master")
+    version("0.9.0", sha256="77d8c9e64947fc9f5cc05bbc6aed7b8a9907871ae26fe11997fd67be90f26008")
     version("0.5.2", sha256="1d7657a4b8ef9aa6d81e49b369c677c584e83269f11710557741d3b3f8fa0a75")
 
     variant("plots", default=True, description="Build with plot support from py-matplotlib")
-    variant("gptune", default=False, description="Build with patches for GPTune")
+    variant("gptune", default=False, description="Build with patches for GPTune", when="@0.5.2")
```

Comment:
```suggestion
    variant("gptune", default=False, description="Build with patches for GPTune")
```
Technically this variant shouldn't even be in here, it should be added to the gptune package. The only package that needs this is gptune, which requires the master branch, which is not 0.5.2, hence the failing tests. Let's leave this alone, maybe @liuyangzhuan can move this patch to the gptune package in a separate PR?

Reply:
@adamjstewart Could you point out how to add a patch file to the py-scikit-optimize subdirectory from modifying the package.py file of gptune?

Reply:
An example from the `gdal` package:
```python
    depends_on("jasper@1.900.1", patches=[patch("uuid.patch")], when="+jasper")                     
```
The `uuid.patch` file, stored alongside the GDAL `package.py`, will be applied to `jasper` when building it as a dep of `gdal`

Reply:
Got it! I will do that in a few days, I'm currently away from my workstation for this week.

Reply:
@adamjstewart Can you close this PR and instead take a look at https://github.com/spack/spack/pull/39447 ?

Reply:
Let's keep "adding a new version of py-scikit-optimize" and "moving the patch to a different package" as separate PRs.